test-case-reporting/app.ts

```diff
@@ -142,12 +143,19 @@ app.post('/report', jsonParser, async (req, res) => {
 });
 
 app.get('/_cron/compute-stats', async (req, res) => {
+  const testCaseStats = new TestCaseStats(db);
   const {count} = req.query;
 
-  const message = `Computing pass/fail % for past ${count} runs`;
-  res.send(message);
+  try {
+    const sampleSize = parseInt(count.toString());
+    await testCaseStats.updateStats(sampleSize);
```

Comment:
If no value is provided, or an invalid value that produces `NaN`, this will explode. Lets sanitize here, and also gate it so that it can't be invoked by any ol' schmuck who tries to DDoS and overload the DB with stat calculation.

```javascript
// This header is set by App Engine when running cron tasks, and is
// stripped out of external requests.
if (!req.header('X-Appengine-Cron') && process.env.NODE_ENV !== 'dev') {
  throw new Error('Attempted external request to a cron endpoint');
}
```

Reply:
(See `amp-github-apps/owners/info-server.js:147` to see this in action)

Reply:
Done.